### ISSUE TITLE
The checkstyle plugin does not accept imports that are only used in JavaDoc comments

### ISSUE THREAD

**Issue by mhdirkse** [ibissource/iaf#3196]
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
If I import a Java class only to reference its simple name in a JavaDoc {@link ...} directive, then the build fails. The checkstyle plugin complains that there is an unused import.

**Reporter**
Martijn Dirkse

**To Reproduce**
1. Take a Java source file. Update the JavaDoc above the class by adding a {@link ...} that points to a class that is not yet imported.
2. Add an import pointing to the referenced class.
3. Run the Maven build.
4. See the error: The checkstyle plugin says the import of the new class is not used.

**Screenshots**

**Configuration**
```
Please provide the configuration of the Pipe or Receiver with the problem.
```

**Input**
```
Please provide an example of the input message. 
Alternatively, provide a Ladybug report.
```

**Environment**
```
Please go to the console of your Frank, click on 'Information' and copy-paste the results here. 
```

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

**Comment by nielsm5** [comment]
This is a configurable setting `processJavadoc` which is turned on by default.
According to CheckStyle it is considered a bad practice to introduce a compile-time dependency for documentation purposes only. See the [CheckStyle UnusedImports Documentation](https://checkstyle.sourceforge.io/config_imports.html#UnusedImports) for more info.

As it improves the readability for developers, we can opt to disable this check. 
For Eclipse it's important that people read our [contributing](
https://github.com/ibissource/iaf/blob/master/CONTRIBUTING.md#let-eclipse-check-javadoc-comments) page, and configure Javadoc processing accordingly.

**Comment by nielsm5** [comment]
I cannot seem to reproduce that this is failing, it appears to always have been enabled.
When and where did you have this issue?

**Comment by mhdirkse** [comment]
I tried to reproduce the issue, but that failed. I temporarily updated Adapter.java to have a JavaDoc pointing to DigesterRule.java (both ibis-adapterframework-core). There was a {@link ...} directive and an import. The build succeeded.

I was curious whether unused imports were detected. They were: the build failed when I removed the {@link ...} in the JavaDoc of Adapter.java.

I think we can close this issue.
